content/toolsAndResources/nge.md

```diff
@@ -0,0 +1,130 @@
+
+The names of the inputs and outputs are taken from the name of the input or output where the wire originated. To help with clarity, however, by clicking on any input or output on a collapsed frame, the parameters panel will display an option to set a custom name. This allows collapsed frames to be more specific about where the port will lead inside the frame when collapsed. Expanding the frame will retain the orignal node positions and wire passing outside the frame. Note that an input or output that was renamed when collapsed will retain the custom name when expanded.
+
+![To make a collapsed frame more user friendly, click on an input or output to change the name of the port.](/img/tools/nge/customInputName.jpg)
+
+### Custom Frames
+
+Frames also enable reuse of parts of your graph elsewhere. If there is a part of a graph that seems like it could be a common operation through a collection of mesh graphs, simply create a frame around the nodes and export the frame using the `Export` button in the parameters panel. Clicking on the `Add...` button under Custom Frames in the node list enables import of the custom frame into the node list. Once imported to the node list, the custom node will be available any time the editor is launched so it only needs to be imported once. Additionally, the frame comments will appear as a tool tip when hovering the frame in the node list to give context to each frame as the custom frame list grows.
+
+From there the custom frame can be dragged into the graph as any other node and will place in collapsed form to reduce the amount of needed space. Custom frames also appear in search results so searching in either the node list or the shortcut node list opened with the spacebar will return custom nodes to make adding them even faster. Once in the graph, it acts like any other frame and can be expanded, collapsed, and arranged in any manner desired. The frames can also be directly edited in the graph or changed completely. In this way, custom frames can act like a template bringing in common collections of nodes to a graph and speeding the process of creating graphs. 
+
+![Exporting a frame allows for import of the frame to the node list while retaining the frame comments to display on hover and is searchable](/img/tools/nge/exportFrameFlow.jpg)
+
+### Wire Elbows
+
+When working with graph editors, one of the ways it can be hard to read comes from the natural process of wires connecting different nodes turning into spaghetti. If it is not clear which path a wire is taking becuase it crosses or overlaps other wires or sits beneath other nodes, comprehension of the graph goes down and it takes longer to understand what the flow is actually doing. Where frames help organize nodes, elbows help organize the wires in a graph. Consider the graph below and note that the overlapping wires can make it hard to discern which wires are flowing to which nodes.

```

Comment:
nit: becuase => because

Reply:
not a nit... a bug. Fixed. I do miss spelling/grammar checking when writing directly in VSCode. Installing a spell checker extension!

Reply:
updated in https://github.com/BabylonJS/Documentation/pull/963/commits/07fc54158e3862125b1824623a58ae435c2dd890